proposals/12238-notebook-component/README.md

```diff
@@ -0,0 +1,295 @@
+<!-- /toc -->
+
+## Summary
+
+This proposal introduces a first-class `@dsl.notebook_component` decorator that lets users build Kubeflow Pipelines
+(KFP) components directly from Jupyter notebooks. The decorator embeds a `.ipynb` file into the component and executes
+it at runtime via `nbconvert`, with parameters injected as a prepended cell. This provides a simple path for
+notebook-centric workflows to run in KFP without requiring separate packaging or bespoke wrappers.
+
+## Motivation
+
+Many users begin experimentation and development in Jupyter notebooks. Turning those notebooks into pipeline components
+currently requires boilerplate: exporting to Python, writing a wrapper function, or managing custom container images. A
+native notebook component:
+
+- Reduces friction to productionize notebook code in pipelines
+- Preserves the notebook as the source of truth while allowing parameterization
+- Avoids extra build steps by embedding notebook content into the component
+
+### Goals
+
+1. Enable defining a component from a `.ipynb` notebook with a single decorator.
+2. Support parameter injection into the notebook at execution time.
+3. Use the existing Python executor to keep compatibility with existing inputs/outputs concepts.
+
+### Non-Goals
+
+1. Notebook validation or linting beyond JSON and structural checks.
+2. Adding a new pipeline IR or backend executor. This reuses the Python executor.
+
+## Proposal
+
+### Baseline Feature For Bundled Assets
+
+This KEP establishes a baseline, generic capability for Python-function components to embed arbitrary files or
+directories directly into a lightweight component. The goal is to support cases where a component needs a small amount
+of read-only assets (configs, scripts, models, notebooks, etc.) without requiring a custom image.
+
+- Add a new decorator argument to `@dsl.component`:
+  - `bundled_artifact_path: Optional[str]` — path to a file or directory on the authoring machine to bundle with the
+    component.
+- Add a new SDK type: `dsl.BundledInput[T]` — a runtime-only input annotation that resolves to an artifact instance of
+  type `T` (e.g. `dsl.Dataset`) whose `path` points to the extracted bundle root.
+  - If a directory is bundled, `.path` points to the extracted directory.
+  - If a single file is bundled, `.path` points to that file.
+
+Example:
+
+```python
+@dsl.component(
+    bundled_artifact_path="assets/config_dir",
+)
+def my_component(cfg: dsl.BundledInput[dsl.Dataset], param: int):
+    # cfg.path is a directory when a directory is bundled; a file path if a file is bundled
+    print(cfg.path)
+```
+
+Execution model (lightweight components):
+
+- At compile time, the file/dir is archived (tar + gzip) and base64-embedded into the ephemeral module.
+- At runtime, the bundle is safely extracted to a temporary location and the `BundledInput[...]` parameter is injected
+  as an artifact with `.path` pointing to the extracted file/dir in a temporary directory.
```

Comment:
Also, would the path be handled in a way that it is considered before all the system paths? Or should it be after all the system paths?